src/main/java/com/google/research/bleth/simulator/Location.java

```diff
@@ -0,0 +1,12 @@
+package com.google.research.bleth.simulator;
+
+/** A location on board. */
+public class Location {
```

Comment:
Consider using AutoValue (https://github.com/google/auto/blob/master/value/userguide/index.md).

Its main advantage for you - in this specific case - might be just the auto-generated toString() (otherwise debugging is a pain). But other cases may also need proper equals() and hashCode() semantics.

Reply:
We consider generating the Location class using protobuf (instead of this implementation), so these methods will be generated as well, and also this struct could be used in JS code (when parsing and visualizing a board state, for example).

Reply:
Protobuf is also OK.